public/scripts/scripts.js

```diff
@@ -102,10 +100,10 @@ function developPlayListStats(allCallSongs, trackBatch){
     }
   }
   console.log(trackBatch);
-  return(trackBatch);
+  return trackBatch;
 };
 
-function idMatcher(identification){
+function idMatcher(identification, playListTracks){
```

Comment:
i think this function broke in the commit that removed the  `playlistTracks` global variable; threading that data through to this function when it's called appears to solve the problem